### ISSUE TITLE
Random crash while flying around

### ISSUE THREAD

**Issue by freeeranger** [jbredwards/Fluidlogged-API#31]
Hey, just got this crash when I was flying around in my modpack with fluidlogged (latest version) installed.

Description: Exception ticking world

java.lang.NoClassDefFoundError: net/minecraft/world/World$2
	at net.minecraft.world.World.func_190524_a(World.java:557)
	at net.minecraft.world.World.func_175685_c(World.java:492)
	at net.minecraft.world.World.func_175722_b(World.java:440)
	at net.minecraft.world.World.markAndNotifyBlock(World.java:381)
	at net.minecraft.world.World.func_180501_a(World.java:361)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:491)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:491)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:491)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:129)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:133)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:491)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.World.func_189507_a(World.java:2682)
	at net.minecraft.world.gen.feature.WorldGenLiquids.func_180709_b(WorldGenLiquids.java:88)
	at net.minecraft.world.biome.BiomeDecorator.func_150513_a(BiomeDecorator.java:352)
	at net.minecraft.world.biome.BiomeDecorator.func_180292_a(BiomeDecorator.java:88)
	at net.minecraft.world.biome.Biome.func_180624_a(Biome.java:225)
	at biomesoplenty.common.world.ChunkGeneratorOverworldBOP.func_185931_b(ChunkGeneratorOverworldBOP.java:565)
	at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1019)
	at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:980)
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
	at net.minecraft.server.management.PlayerChunkMapEntry.func_187268_a(PlayerChunkMapEntry.java:126)
	at net.minecraft.server.management.PlayerChunkMap.func_72693_b(PlayerChunkMap.java:226)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:227)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:756)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:279)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Unknown Source)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at net.minecraft.world.World.func_190524_a(World.java:557)
	at net.minecraft.world.World.func_175685_c(World.java:492)
	at net.minecraft.world.World.func_175722_b(World.java:440)
	at net.minecraft.world.World.markAndNotifyBlock(World.java:381)
	at net.minecraft.world.World.func_180501_a(World.java:361)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:491)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:491)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:491)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:129)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:133)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:491)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.World.func_189507_a(World.java:2682)
	at net.minecraft.world.gen.feature.WorldGenLiquids.func_180709_b(WorldGenLiquids.java:88)
	at net.minecraft.world.biome.BiomeDecorator.func_150513_a(BiomeDecorator.java:352)
	at net.minecraft.world.biome.BiomeDecorator.func_180292_a(BiomeDecorator.java:88)
	at net.minecraft.world.biome.Biome.func_180624_a(Biome.java:225)
	at biomesoplenty.common.world.ChunkGeneratorOverworldBOP.func_185931_b(ChunkGeneratorOverworldBOP.java:565)
	at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1019)
	at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:980)
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
	at net.minecraft.server.management.PlayerChunkMapEntry.func_187268_a(PlayerChunkMapEntry.java:126)
	at net.minecraft.server.management.PlayerChunkMap.func_72693_b(PlayerChunkMap.java:226)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:227)

**Comment by jbredwards** [comment]
Unfortunately I can't replicate this. Can you send the full log? NoClassDefFoundError is usually thrown as the result of an earlier error.

**Comment by Shazuli** [comment]
I got the same exact crash too when flying around.
<details>
  <summary>crash-2021-08-21_15.47.05-server.txt</summary>

```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  CreativePatchingLoader (CreativeCore_v1.10.61_mc1.12.2.jar)
  SSLoadingPlugin (SereneSeasons-1.12.2-1.2.18-universal.jar)
  CorePlugin (SmoothFont-mc1.12.2-2.1.2.jar)
  TransformerLoader (OpenComputers-MC1.12.2-1.7.5.192.jar)
  MixinBooter (mixinbooter-2.0.jar)
  Thaumcraft Research Patcher Core Plugin (ThaumcraftResearchPatcher-1.12.2-1.1.2.jar)
  Quark Plugin (Quark-r1.6-179.jar)
  PerformantLoadingPlugin (performant-1.12.2-1.10.jar)
  CTMCorePlugin (CTM-MC1.12.2-1.0.2.31.jar)
  Fluidlogged Plugin (Fluidlogged-API-v1.6.2-mc1.12.2.jar)
  LoadingPlugin (Bloodmoon-MC1.12.2-1.5.3.jar)
  GTCELoadingPlugin (gregtech-1.12.2-1.16.1.753.jar)
  PhosphorFMLLoadingPlugin (phosphor-1.12.2-0.2.6+build50-universal.jar)
  ParticleCullingLoadingPlugin (particleculling-1.12.2-v1.0.1.jar)
  BlurPlugin (Blur-1.0.4-14.jar)
  ItemPatchingLoader (ItemPhysic_Full_1.4.34_mc1.12.2.jar)
  FMLPlugin (Mirage-2.0.3-rc3.jar)
  LoadingPlugin (HardcoreDarkness-MC1.12.2-2.0.jar)
  SmoothWaterCore (SmoothWater-0.1.4.jar)
  Thaumic Augmentation Core Plugin (ThaumicAugmentation-1.12.2-2.1.4.jar)
  EntityCullingPlugin (EntityCulling-1.12.2-2.0.5.jar)
  DynamicSurroundingsCore (DynamicSurroundings-core-1.12.2-3.5.4.3.jar)
  AdvancedRocketryPlugin (AdvancedRocketry-1.12.2-2.0.0-248-universal.jar)
  LoadingPlugin (ChunkAnimator-MC1.12-1.2.jar)
Contact their authors BEFORE contacting forge

// On the bright side, I bought you a teddy bear!

Time: 8/21/21 3:47 PM
Description: Exception ticking world

java.lang.NoClassDefFoundError: net/minecraft/world/World$2
	at net.minecraft.world.World.func_190524_a(World.java:557)
	at net.minecraft.world.World.func_175685_c(World.java:493)
	at net.minecraft.world.World.func_175722_b(World.java:440)
	at net.minecraft.world.World.markAndNotifyBlock(World.java:381)
	at net.minecraft.world.World.func_180501_a(World.java:361)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:492)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:133)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:492)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:133)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:492)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:492)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.World.func_189507_a(World.java:2682)
	at net.minecraft.world.gen.feature.WorldGenLiquids.func_180709_b(WorldGenLiquids.java:88)
	at net.minecraft.world.biome.BiomeDecorator.func_150513_a(BiomeDecorator.java:352)
	at net.minecraft.world.biome.BiomeDecorator.func_180292_a(BiomeDecorator.java:88)
	at net.minecraft.world.biome.Biome.func_180624_a(Biome.java:225)
	at net.minecraft.world.gen.ChunkGeneratorOverworld.func_185931_b(ChunkGeneratorOverworld.java:464)
	at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1019)
	at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:990)
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
	at net.minecraft.server.management.PlayerChunkMapEntry.func_187268_a(PlayerChunkMapEntry.java:126)
	at net.minecraft.server.management.PlayerChunkMap.func_72693_b(SourceFile:147)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:227)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:756)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:185)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Thread.java:748)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at net.minecraft.world.World.func_190524_a(World.java:557)
	at net.minecraft.world.World.func_175685_c(World.java:493)
	at net.minecraft.world.World.func_175722_b(World.java:440)
	at net.minecraft.world.World.markAndNotifyBlock(World.java:381)
	at net.minecraft.world.World.func_180501_a(World.java:361)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:492)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:133)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:492)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.block.BlockDynamicLiquid.func_176375_a(BlockDynamicLiquid.java:175)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:133)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:492)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.WorldServer.func_175654_a(WorldServer.java:551)
	at net.minecraft.world.WorldServer.func_175684_a(WorldServer.java:531)
	at net.minecraft.block.BlockDynamicLiquid.func_176213_c(BlockDynamicLiquid.java:306)
	at net.minecraft.world.chunk.Chunk.func_177436_a(Chunk.java:614)
	at net.minecraft.world.World.func_180501_a(World.java:343)
	at net.minecraft.world.World.func_175656_a(World.java:425)
	at git.jbredwards.fluidlogged_api.asm.ASMHooks.flowInto(ASMHooks.java:492)
	at net.minecraft.block.BlockDynamicLiquid.func_180650_b(BlockDynamicLiquid.java:138)
	at net.minecraft.world.World.func_189507_a(World.java:2682)
	at net.minecraft.world.gen.feature.WorldGenLiquids.func_180709_b(WorldGenLiquids.java:88)
	at net.minecraft.world.biome.BiomeDecorator.func_150513_a(BiomeDecorator.java:352)
	at net.minecraft.world.biome.BiomeDecorator.func_180292_a(BiomeDecorator.java:88)
	at net.minecraft.world.biome.Biome.func_180624_a(Biome.java:225)
	at net.minecraft.world.gen.ChunkGeneratorOverworld.func_185931_b(ChunkGeneratorOverworld.java:464)
	at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1019)
	at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:990)
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
	at net.minecraft.server.management.PlayerChunkMapEntry.func_187268_a(PlayerChunkMapEntry.java:126)
	at net.minecraft.server.management.PlayerChunkMap.func_72693_b(SourceFile:147)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:227)

-- Affected level --
Details:
	Level name: New World
	All players: 1 total; [EntityPlayerMP['simon_kungen'/789, l='New World', x=-3427.94, y=160.30, z=-720.46]]
	Chunk stats: ServerChunkCache: 1354 Drop: 0
	Level seed: -1291364878882627603
	Level generator: ID 09 - quark_realistic, ver 0. Features enabled: true
	Level generator options: 
	Level spawn location: World: (-72,64,256), Chunk: (at 8,4,0 in -5,16; contains blocks -80,0,256 to -65,255,271), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Level time: 12652 game time, 12652 day time
	Level dimension: 0
	Level storage version: 0x04ABD - Anvil
	Level weather: Rain time: 64705 (now: false), thunder time: 19842 (now: false)
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: true
Stacktrace:
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:756)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:185)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
	at java.lang.Thread.run(Thread.java:748)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Linux (amd64) version 5.4.0-80-generic
	Java Version: 1.8.0_291, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 1235247120 bytes (1178 MB) / 8205107200 bytes (7825 MB) up to 8205107200 bytes (7825 MB)
	JVM Flags: 2 total; -Xms4096m -Xmx8064m
	IntCache: cache: 12, tcache: 94, allocated: 0, tallocated: 0
	FML: ~~ERROR~~ NoClassDefFoundError: net/minecraftforge/common/util/TextTable$1
	Loaded coremods (and transformers): 
CreativePatchingLoader (CreativeCore_v1.10.61_mc1.12.2.jar)
  
SSLoadingPlugin (SereneSeasons-1.12.2-1.2.18-universal.jar)
  sereneseasons.asm.transformer.EntityRendererTransformer
  sereneseasons.asm.transformer.WorldTransformer
CorePlugin (SmoothFont-mc1.12.2-2.1.2.jar)
  bre.smoothfont.asm.Transformer
TransformerLoader (OpenComputers-MC1.12.2-1.7.5.192.jar)
  li.cil.oc.common.asm.ClassTransformer
MixinBooter (mixinbooter-2.0.jar)
  
Thaumcraft Research Patcher Core Plugin (ThaumcraftResearchPatcher-1.12.2-1.1.2.jar)
  thecodex6824.tcresearchpatcher.Transformer
Quark Plugin (Quark-r1.6-179.jar)
  vazkii.quark.base.asm.ClassTransformer
PerformantLoadingPlugin (performant-1.12.2-1.10.jar)
  
CTMCorePlugin (CTM-MC1.12.2-1.0.2.31.jar)
  team.chisel.ctm.client.asm.CTMTransformer
Fluidlogged Plugin (Fluidlogged-API-v1.6.2-mc1.12.2.jar)
  git.jbredwards.fluidlogged_api.asm.swapper.ASMSwapper
  git.jbredwards.fluidlogged_api.asm.ASMHandler$Transformer
LoadingPlugin (Bloodmoon-MC1.12.2-1.5.3.jar)
  lumien.bloodmoon.asm.ClassTransformer
GTCELoadingPlugin (gregtech-1.12.2-1.16.1.753.jar)
  gregtech.common.asm.GTCETransformer
PhosphorFMLLoadingPlugin (phosphor-1.12.2-0.2.6+build50-universal.jar)
  
ParticleCullingLoadingPlugin (particleculling-1.12.2-v1.0.1.jar)
  
BlurPlugin (Blur-1.0.4-14.jar)
  com.tterrag.blur.BlurTransformer
ItemPatchingLoader (ItemPhysic_Full_1.4.34_mc1.12.2.jar)
  com.creativemd.itemphysic.ItemTransformer
FMLPlugin (Mirage-2.0.3-rc3.jar)
  com.elytradev.mirage.asm.RenderGlobalTransformer
  com.elytradev.mirage.asm.ChunkRenderContainerTransformer
LoadingPlugin (HardcoreDarkness-MC1.12.2-2.0.jar)
  lumien.hardcoredarkness.asm.ClassTransformer
SmoothWaterCore (SmoothWater-0.1.4.jar)
  pl.asie.smoothwater.SmoothWaterTransformer
Thaumic Augmentation Core Plugin (ThaumicAugmentation-1.12.2-2.1.4.jar)
  thecodex6824.thaumicaugmentation.core.TATransformer
EntityCullingPlugin (EntityCulling-1.12.2-2.0.5.jar)
  meldexun.entityculling.plugin.ClassTransformer
DynamicSurroundingsCore (DynamicSurroundings-core-1.12.2-3.5.4.3.jar)
  org.orecruncher.dsurround.asm.Transformer
AdvancedRocketryPlugin (AdvancedRocketry-1.12.2-2.0.0-248-universal.jar)
  zmaster587.advancedRocketry.asm.ClassTransformer
LoadingPlugin (ChunkAnimator-MC1.12-1.2.jar)
  lumien.chunkanimator.asm.ClassTransformer
	GL info: ~~ERROR~~ RuntimeException: No OpenGL context found in the current thread.
	AE2 Version: stable rv6-stable-7 for Forge 14.23.5.2768
	forestry : Modules have been disabled in the config: TechReborn, Charcoal, Crate, Mystical Agriculture, Worktable, Energy, Mail, ImmersiveEngineering, Better With Mods, Factory, Actually Additions, Roots, EnderIO
	AE2 Integration: IC2:OFF, RC:OFF, MFR:OFF, Waila:ON, InvTweaks:OFF, JEI:OFF, Mekanism:OFF, OpenComputers:ON, THE_ONE_PROBE:OFF, TESLA:OFF, CRAFTTWEAKER:ON
	Profiler Position: N/A (disabled)
	Player Count: 1 / 8; [EntityPlayerMP['simon_kungen'/789, l='New World', x=-3427.94, y=160.30, z=-720.46]]
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'fml,forge'
```
</details>
I got this crash when flying around just like @freerangerstudios. I was running a pretty heavy modpack and was generating a lot of terrain, so the integrated server was running pretty hard.
When I tried to join the world again and flew over the exact same place everything was fine. I was flying over a ocean, so maybe a block got bad data when attempting to fluidlogg it? I have no idea what block that could be, maybe a Rail block from an open Mineshaft?

**Comment by embeddedt** [comment]
@Shazuli The full log (not just the crash) would be useful for this, as this error is usually caused by a Mixin or ASM transform failing to apply (and those errors are not reported in the crash.txt file).

**Comment by Shazuli** [comment]
Alright. Sorry for the late reply, been really busy with school and stuff.
I managed to find the log (thought it got deleted every time):
[2021-08-21-1.log.gz](https://github.com/jbredwards/Fluidlogged-API/files/7083000/2021-08-21-1.log.gz)

Pretty sure that's the right one. The date and time match so it should be the right one. From what I could read from the logs myself the last warning before it crashes (line `7007`) is about a Railcraft block. Which is strange because I don't have Railcraft installed in this instance. I tried installing it before but the game didn't launch then so the world has never had Railcraft installed when created.

**Comment by embeddedt** [comment]
The strange thing is that in this crash log the game doesn't appear to reach the main menu, but instead crashes while syncing registries. Additionally, Railcraft was definitely installed at the time of the crash (line 7157).

Possibly it's the wrong log?

**Comment by Shazuli** [comment]
Yeah, you're right. Maybe that's the log from when I tried to run the game with Railcraft? It is the earliest log from that day (08-21) but the log is around 30 minutes later. Did it fail to generate the log when it crashed?

**Comment by embeddedt** [comment]
It's possible. The exact error appears to be a StackOverflowError, but this isn't reported correctly due to the fact that the stack has overflown. :wink:

@jbredwards I have just run into a similar stacktrace while developing another 1.12 mod. It seems that this occurs when there is a recursion issue in one of the "neighbor change"-related methods.

**Comment by Shazuli** [comment]
Alright, managed to encounter the crash again @jbredwards. This time I managed to save the log on [this link](https://paste.ee/p/J3ZaQ).
Attaching the crash log too:
[crash-2021-10-19_12.46.21-server.txt](https://github.com/jbredwards/Fluidlogged-API/files/7372526/crash-2021-10-19_12.46.21-server.txt)

Hopefully you'll be able to narrow down the cause now.

**Comment by jbredwards** [comment]
This will definitely help ty. In the meantime it may be a good idea to remove the MAGE mod, since fluidlogged api has its smoothwater fix built-in.

**Comment by jbredwards** [comment]
fixed in v1.7
